### PR TITLE
bsdlib: Update BSD library for nrf9160

### PR DESCRIPTION
**Pull request by lemrey**
* Added AGPS support to GNSS socket driver.
* Added support for GNSS power save modes.
* Added support for deleting stored GPS data.
* Changed NRF_CONFIG_NMEA* define names to NRF_GNSS_NMEA* for alignment.

Library revision: 18728427ba029f6ee6e0379286d332e967471cb8

CC: @gbakke

### CODE REVIEW

**Comment by rlubos** [comment]
Please bump the library version and update the changelog file.

**Comment by lemrey** [comment]
Is there any other place to bump the version other than the changelog?

**Comment by rlubos** [comment]
I don't think so, no.

**Comment by eibe85** [comment]
I will run the GPS test we have on this and the LTE and NB-IOT tests I guess I do not need to run the invalid test cases as I guess there are no updates to errnos?

**Comment by evenl** [comment]
GPS entry is ok, I don't think it make sense to list data model supported

**Comment by eibe85** [comment]
Can you list what is the correct masks for:
NRF_CONFIG_NMEA_GSV_MASK
NRF_CONFIG_NMEA_GSA_MASK
NRF_CONFIG_NMEA_GLL_MASK
NRF_CONFIG_NMEA_GGA_MASK
NRF_CONFIG_NMEA_RMC_MASK

**Comment by eibe85** [comment]
The test fail on:
    retval = nrf_setsockopt(fd,
                NRF_SOL_GNSS,
                NRF_SO_GNSS_START,
                NULL,
                0);

    zassert_true(retval == 0, "Failed to start GPS. Errno: %d \n", errno);
in init_app()

**Comment by eibe85** [comment]
After the update to use:
    uint32_t delete_mask = 0;
    retval = nrf_setsockopt(fd,
                NRF_SOL_GNSS,
                NRF_SO_GNSS_START,
                &delete_mask,
                sizeof(delete_mask));
GPS tests are passing. Have also run the LTE test suite and that to is passing.

**Comment by eibe85** [comment]
Pleas also add the above API change to the release notes

**Comment by rlubos** [comment]
@lemrey Could you update the release notes as requested, and fix the commit title?

**Comment by lemrey** [comment]
Updated changelog and git commit message.

File: bsdlib/include/nrf_socket.h [hunk 1]
```diff
@@ -184,15 +185,36 @@ typedef int32_t ssize_t;
...
-#define NRF_CONFIG_NMEA_GSV_MASK     8  /**< Enables Satellites in view. */
-#define NRF_CONFIG_NMEA_RMC_MASK     16 /**< Enables Recommended minimum specific GPS/Transit data. */
+#define NRF_GNSS_NMEA_GGA_MASK     1  /**< Enables Global Positioning System Fix Data. */
+#define NRF_GNSS_NMEA_GLL_MASK     2  /**< Enables Geographic Position Latitude/Longitude and time. */
+#define NRF_GNSS_NMEA_GSA_MASK     4  /**< Enables DOP and active satellites. */
+#define NRF_GNSS_NMEA_GSV_MASK     8  /**< Enables Satellites in view. */
+#define NRF_GNSS_NMEA_RMC_MASK     16 /**< Enables Recommended minimum specific GPS/Transit data. */
+/** @} */
+
+/**@defgroup nrf_gnss_psm_modes Values indicating the power saving mode used in the GNSS module
```
Comment (evenl) [inline bsdlib/include/nrf_socket.h]: This sentence sounds a bit strange. 'indicating' sounds like the application is not an active part of this data.

File: bsdlib/include/nrf_socket.h [hunk 2]
```diff
@@ -184,15 +185,36 @@ typedef int32_t ssize_t;
 #define NRF_GNSS_SV_FLAG_UNHEALTHY    8 /**< Indicate that the satellite is unhealthy. */
 /**@} */
 
+/**@defgroup nrf_gnss_agps_data_type Flags indicating which AGPS assistance data set is written to the GPS module
```
Comment (evenl) [inline bsdlib/include/nrf_socket.h]: These values are are not flags, but enumerator values.

File: bsdlib/include/nrf_socket.h (hunk 1 above)
Comment (b-gent) [reply bsdlib/include/nrf_socket.h]:
```suggestion
/**@defgroup nrf_gnss_psm_modes Values setting the power saving mode used in the GNSS module
```

Does that make more sense?

File: bsdlib/include/nrf_socket.h (hunk 2 above)
Comment (b-gent) [reply bsdlib/include/nrf_socket.h]:
```suggestion
/**@defgroup nrf_gnss_agps_data_type Values setting which AGPS assistance data set is written to the GPS module
```

Comment (b-gent) [reply bsdlib/include/nrf_socket.h]: Seems like I just commented on a merged PR :)

**Pull request merged**